### ISSUE TITLE
[question] Change camera position via StructureFromKnownPoses

### ISSUE THREAD

**Issue by raphael2692** [alicevision/meshroom#807]
I followed the suggestions in issues #725, #788, #787 and tried to use the SFKP node to impose the known position of my cameras. However, changing rotation and center parameters in the 'cameras.sfm' node does not produce any effect (edited cameras look always in the same position, despite extreme values like center [1000, 1000, 1000].

**Comment by natowi** [comment]
![pr](https://user-images.githubusercontent.com/8138537/75277657-b5205880-5808-11ea-8a9e-a264d44bbd4d.JPG) 

**Here I changed the camera from**

```
"center": [
                        "-0.019798363554132201",
                        "0.002777069028680352",
                        "0.0027687477898859897"
                    ]
```

**to**

```
"center": [
                        "-0.19000000000000000",
                        "0.027000000000000000",
                        "0.027000000000000000"
                    ]
```

**Comment by raphael2692** [comment]
Thanks @natowi , I managed to move the camera in a limited value range.  Can you also help me understanding what kind of metrics are in use in the rotation matrix and in the 'center' array, so I can translate my rig baseline to it?

Also, do you think that after imposing fixed intrisics and camera position this way I would achive that scanned models always come at the same scale? 

Thanks! 😊

**Comment by raphael2692** [comment]
On a side note, I tried running the full pipe with the fixed camera input but I got stuck at depthmap filter node. Also, Only 3 dephtmap out of 6 are computed (while with default settings they are all computed and I can run all the pipeline). Notice that I used the node with the cameras.sfm in input but without altering it. 

![image](https://user-images.githubusercontent.com/24436489/75336125-dde93200-588a-11ea-8811-989918891bf3.png)

**Comment by natowi** [comment]
> Can you also help me understanding what kind of metrics are in use in the rotation matrix and in the 'center' array, so I can translate my rig baseline to it?
> 
> Also, do you think that after imposing fixed intrisics and camera position this way I would achive that scanned models always come at the same scale?

You need a reference (known points, scale) for that.
At the moment there is only the SfMTransform node.
With the node you can apply a given transformation camera as the origin of the coordinate system  or rescale the scene based on the bounding box of CCTAG markers.
https://github.com/alicevision/AliceVision/issues/402

> got stuck at depthmap filter node. Also, Only 3 dephtmap out of 6 are computed

Does the default graph work with the images?

**Comment by raphael2692** [comment]
> Does the default graph work with the images?

yes it does.

> At the moment there is only the SfMTransform node.

I am aware of that, but it is unsuitable for my use, as I cannot place markers. I thought that by fixing camera position and baseli I would achieve the same. Isn't that so?

**Comment by natowi** [comment]
SfMTransform 
**Transformation Method: transformation**

Align [X,Y,Z] to +Y-axis, rotate around Y by R deg, scale by S; syntax: X,Y,Z;R;S (all five parameters are required)
This allows the user to align and scale the point cloud by explicitly specifying the scale and "up" vector [X,Y,Z] in the point cloud's reference system. The rotation is such that the specified [X,Y,Z] vector is aligned with [0,1,0] after the transformation.

Load your point cloud or model in Meshlab and read off the transformation parameters, then you can use them in this node. For a fixed rig the parameters should stay the same.

**Comment by raphael2692** [comment]
Thanks for your reply @natowi, I will check out that node. However prior to that, scale recostruction should be regular. I found that with the same camera (fixed position, syncronized trigger + flash) the scale factor largely fluctuate. Is there a parameter I can use to fix this?

![image](https://user-images.githubusercontent.com/24436489/75566725-7da3ed00-5a50-11ea-87e1-4f3be4fbd4f0.png)

**Comment by natowi** [comment]
It´s a mystery to me, too...

Another thing you can try is the SfMAlignment node https://github.com/alicevision/meshroom-manual/blob/WIP-Update-node-reference/source/node-reference/nodes/SfMAlignment.rst

You could export and scale a SfM file and use it as reference for alignment. (Yet I did not test this myself)

**Comment by fabiencastan** [comment]
@raphael2692 

> I thought that by fixing camera position and baseli I would achieve the same. Isn't that so?

Yes, as said by @natowi you could use the SfMAlignment node to align multiple reconstruction taken from a fixed set of cameras.

**Comment by raphael2692** [comment]
thanks for the reply @natowi and @fabiencastan . I was wondering, is SfMAlignment capable of correctly scaling *different* object placed around the same spot in the scene? Also, is the node influenced by the number of features extracted/light conditions from the fixed cameras dataset?

**Comment by fabiencastan** [comment]
SfMAlignment can do the alignment based on common Cameras (if you have a fixed cameras rig) or common Markers. So it does not care about the shooting conditions, number of features, etc.

**Comment by untbu** [comment]
> On a side note, I tried running the full pipe with the fixed camera input but I got stuck at depthmap filter node. Also, Only 3 dephtmap out of 6 are computed (while with default settings they are all computed and I can run all the pipeline). Notice that I used the node with the cameras.sfm in input but without altering it.

I had the same issue because I copied the cameraInit.sfm file and edited the copied file. When I replaced the existing file and used the default connection from CameraInit to FeatureExtraction, it worked.
